src/library-authoring/library-detail/LibraryAuthoringPage.jsx

```diff
@@ -0,0 +1,435 @@
+          size="lg"
+          onClick={() => setShowDeleteModal(true)}
+        >
+          <FontAwesomeIcon icon={faTrashAlt} />
+        </Button>
+      </Navbar.Collapse>
+    </Navbar>
+    <Modal
+      open={showDeleteModal}
+      title={intl.formatMessage(messages['library.detail.block.delete.modal.title'])}
+      onClose={() => setShowDeleteModal(false)}
+      body={(
+        <div>
+          <p>
+            {intl.formatMessage(messages['library.detail.block.delete.modal.body'])}
+          </p>
+        </div>
+      )}
+      buttons={[
+        <Button
+          onClick={() => props.deleteLibraryBlock({ blockId: block.id })}
+        >
+          {intl.formatMessage(commonMessages['library.common.forms.button.yes'])}
+        </Button>,
+      ]}
+    />
+    {showPreviews && (
+      <Card>
+        <Card.Body>
+          <LibraryBlock getHandlerUrl={getHandlerUrl} view={view} key={previewKey} />
+        </Card.Body>
+      </Card>
+    )}
+  </>
+);
+
+BlockPreviewBase.defaultProps = {
+  view: null,
+};
+
+BlockPreviewBase.propTypes = {
+  intl: intlShape.isRequired,
+  block: libraryBlockShape.isRequired,
+  library: libraryShape.isRequired,
+  getHandlerUrl: PropTypes.func.isRequired,
+  view: blockViewShape,
+  canEdit: PropTypes.bool.isRequired,
+  showPreviews: PropTypes.bool.isRequired,
+  showDeleteModal: PropTypes.bool.isRequired,
+  setShowDeleteModal: PropTypes.func.isRequired,
+  deleteLibraryBlock: PropTypes.func.isRequired,
+  previewKey: PropTypes.string.isRequired,
+};
+
+export const BlockPreview = injectIntl(BlockPreviewBase);
+
+/**
+ * BlockPreviewContainerBase
+ * Container component for the BlockPreview cards.
+ * Handles the fetching of the block view and metadata.
+ */
+const BlockPreviewContainerBase = ({
+  intl, getHandlerUrl, block, blockView, blocks, showPreviews, library, ...props
+}) => {
+  useEffect(() => {
+    props.initializeBlock({
+      blockId: block.id,
+    });
+  }, []);
+  useEffect(() => {
+    if (!blocks[block.id] || !showPreviews) {
+      return;
+    }
+    if (blocks[block.id].metadata === null) {
+      props.fetchLibraryBlockMetadata({ blockId: block.id });
+    }
+    if (blocks[block.id].view === null) {
+      props.fetchLibraryBlockView({
+        blockId: block.id,
+        viewSystem: XBLOCK_VIEW_SYSTEM.Studio,
+        viewName: 'student_view',
+      });
+    }
+  }, [blocks[block.id], showPreviews]);
+
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  // Need to force the iframe to be different if navigating away. Otherwise landing on the edit page
+  // will show the student view, and navigating back will show the edit view in the block list. React is smart enough
+  // to guess these iframes are the same between routes and will try to preserve rather than rerender, but that works
+  // against us here. Setting an explicit key prevents it from matching the two.
+  const previewKey = useMemo(() => `${uuid4()}`, [block.id]);
```

Comment:
This explains why I could not get iframes to refresh - sometimes.  TIL how to fix it.  +1 again.